python-sdk/actionml/__init__.py

```diff
@@ -0,0 +1,634 @@
+        Asynchronously create an event.
+        :param event_id: 
+        :param event: event name. type str.
+        :param entity_type: entity type. It is the namespace of the entityId and
+          analogous to the table name of a relational database. The entityId must be
+          unique within same entityType. type str.
+        :param entity_id: entity id. *entity_type-entity_id* becomes the unique
+          identifier of the entity. For example, you may have entityType named user,
+          and different entity IDs, say 1 and 2. In this case, user-1 and user-2
+          uniquely identifies entities. type str
+        :param target_entity_type: target entity type. type str.
+        :param target_entity_id: target entity id. type str.
+        :param properties: a custom dict associated with an event. type dict.
+        :param event_time: the time of the event. type datetime, must contain
+          timezone info.
+        :param creation_time:
+        :returns:
+          AsyncRequest object. You can call the get_response() method using this
+          object to get the final results or status of this asynchronous request.
+        """
+        data = {
+            "eventId": event_id,
+            "event": event,
+            "entityType": entity_type,
+            "entityId": entity_id,
+        }
+
+        if target_entity_type is not None:
+            data["targetEntityType"] = target_entity_type
+
+        if target_entity_id is not None:
+            data["targetEntityId"] = target_entity_id
+
+        if properties is not None:
+            data["properties"] = properties
+
+        data["eventTime"] = time_to_string_if_valid(event_time)
+
+        data["creationTime"] = time_to_string_if_valid(creation_time)
+
+        request = AsyncRequest("POST", self.path, **data)
+        request.set_response_handler(self._create_response_handler)
+        self._connection.make_request(request)
+        return request
+
+    def create(self, event_id: str, event: str, entity_type: str, entity_id: str,
+               target_entity_type: str = None, target_entity_id: str = None, properties: object = None,
+               event_time: datetime = None, creation_time: datetime = None):
+        """Synchronously (blocking) create an event."""
+        return self.async_create(event_id, event, entity_type, entity_id,
+                                 target_entity_type, target_entity_id, properties,
+                                 event_time, creation_time).get_response()
+
+    def async_get(self, event_id: str):
+        """
+        Asynchronously get an event from PIO Kappa Server.
+        :param event_id: event id returned by the EventServer when creating the event.
+        :returns: AsyncRequest object.
+        """
+        request = AsyncRequest("GET", self._add_segment(event_id))
+        request.set_response_handler(self._ok_response_handler)
+        self._connection.make_request(request)
+        return request
+
```

Comment:
This will not be implemented in v0.1.0 of pio-kappa

Many of these will not be implemented (though they will eventually) Stubbing them as not implemented if OK for now.

Only the CLI mentioned in the admin docs CLI should be implemented fully.